packages/@sfx/sayt-driver-plugin/test/unit/common/sayt-driver-plugin.test.ts

```diff
@@ -227,59 +236,280 @@ describe('Sayt Driver Plugin', () => {
       const callbackReturn = ['a', 'b'];
       autocompleteCallback.returns(callbackReturn);
 
-      const returnValue = driver.sendSaytApiRequest(saytDataPayload);
+      const returnValue = driver.sendAutocompleteApiRequest(saytDataPayload);
 
       return expect(returnValue).to.eventually.deep.equal(callbackReturn);
     });
   });
 
-  describe('fetchSaytData()', () => {
+  describe('fetchAutocompleteTerms()', () => {
     let dispatchEvent;
     let results;
     let searchbox;
-    let sendSaytApiRequest;
+    let sendAutocompleteApiRequest;
 
     beforeEach(() => {
       dispatchEvent = dom_events.dispatchEvent = spy();
       driver.core = {
         dom_events,
       };
       results =  { a: 'b' };
-      sendSaytApiRequest = stub(driver, 'sendSaytApiRequest');
+      sendAutocompleteApiRequest = stub(driver, 'sendAutocompleteApiRequest');
+      searchbox = 'some-searchbox-id';
+    });
+
+    it('should call sendAutocompleteApiRequest with query from event and valid config', () => {
+      sendAutocompleteApiRequest.resolves(results);
+
+      driver.fetchAutocompleteTerms(saytDataPayload);
+
+      expect(sendAutocompleteApiRequest).to.be.calledWith(query, config);
     });
 
-    it('should not throw with undefined config', () => {
-      sendSaytApiRequest.resolves(results);
-      const callSayt = () => { driver.fetchSaytData(saytDataUndefinedConfig) };
+    it('should dispatch the response through the events plugin', () => {
+      sendAutocompleteApiRequest.resolves(results);
+
+      driver.fetchAutocompleteTerms(saytDataPayload);
 
-      expect(callSayt).to.not.throw();
+      return expect(Promise.resolve(dispatchEvent))
+        .to.be.eventually.calledOnceWith(driver.autocompleteResponseEvent, { results, searchbox });
     });
 
-    it('should call sendSaytApiRequest with query from event and valid config', () => {
-      sendSaytApiRequest.resolves(results);
+    it('should send an error in an event if the API request fails', () => {
+      const error = new Error('test error');
+      sendAutocompleteApiRequest.rejects(error);
+
+      driver.fetchAutocompleteTerms(saytDataPayload);
 
-      driver.fetchSaytData(saytDataPayload);
+      return expect(Promise.resolve(dispatchEvent))
+        .to.be.eventually.calledOnceWith(driver.autocompleteErrorEvent, error);
+    });
+  });
+
+  describe('fetchProductData', () => {
```

Comment:
For consistency, add `()` to the end of the property name. This applies to the rest of the file as well.

```suggestion
  describe('fetchProductData()', () => {
```